apm-agent-core/src/main/java/co/elastic/apm/agent/bci/bytebuddy/CustomElementMatchers.java

```diff
@@ -87,18 +87,19 @@ public boolean matches(@Nullable ClassLoader target) {
     private static boolean canLoadClass(@Nullable ClassLoader target, String className) {
         boolean result;
         try {
-            Class<?> clazz = Class.forName(className, false, target);
+            if (target == null) {
+                target = ClassLoader.getSystemClassLoader();
```

Comment:
```suggestion
                URL resource = String.class.getResource("/" + className.replace('.', '/') + ".class");
```

What you did will return `true` for the bootstrap class loader and any class loaded from the system classpath, which means this filter will be inefficient for lots of classes. 
A false positive is not so bad here (just a matter of startup time), so we can live with that. However, what I suggested will behave better in some JVMs (at least openjdk 10+)